projects/igniteui-angular/src/lib/date-picker/date-picker.component.ts

```diff
@@ -47,57 +47,213 @@ let NEXT_ID = 0;
+     *```
+     */
     @Input() public weekStart: WEEKDAYS | number = WEEKDAYS.SUNDAY;
 
+    /**
+     *Returns the format options of the `IgxDatePickerComponent`.
+     *```typescript
+     *@ViewChild("MyDatePicker")
+     *public datePicker: IgxDatePickerComponent;
+     *ngAfterViewInit(){
+     *    let formatOptions = this.datePicker.formatOptions;
+     *}
+     *```
+     */
     @Input()
     public get formatOptions(): object {
         return this._formatOptions;
     }
+    /**
+     *Sets the format options of the `IgxDatePickerComponent`.
+     *```typescript
+     *public Options;
+     *@ViewChild("MyDatePicker")
+     *public datePicker: IgxDatePickerComponent;
+     *ngAfterViewInit(){
+     *    this.Options = {
+     *        day: "numeric",
+     *        month: "long",
+     *        weekday: "long",
+     *        year: "numeric"
+     *    }
+     *this.datePicker.formatOptions = this.Options;
+     *}
+     *```
+     */
     public set formatOptions(formatOptions: object) {
         this._formatOptions = Object.assign(this._formatOptions, formatOptions);
     }
 
+    /**
+     *Returns the format views of the `IgxDatePickerComponent`.
+     *```typescript
+     *@ViewChild("MyDatePicker")
+     *public datePicker: IgxDatePickerComponent;
+     *ngAfterViewInit(){
+     *    let formatViews = this.datePicker.formatViews;
+     *}
+     *```
+     */
     @Input()
     public get formatViews(): object {
         return this._formatViews;
     }
+
+    /**
+     *Sets the format views of the `IgxDatePickerComponent`.
+     *```typescript
+     *public allViews;
+     *@ViewChild("MyDatePicker")
+     *public datePicker: IgxDatePickerComponent;
+     *ngAfterViewInit(){
+     *    this.Views = {day:false, month: false, year:false};
+     *    this.datePicker.formatViews = this.Views;
+     *}
+     *```
+     */
     public set formatViews(formatViews: object) {
         this._formatViews = Object.assign(this._formatViews, formatViews);
     }
 
+    /**
+     *An @Input proeprty that sets the orientation of the `IgxDatePickerComponent` header.
+     *```html
+     *<igx-datePicker [vertical]="'true'" cancelButtonLabel="cancel" todayButtonLabel="today"></igx-datePicker>
+     *```
+     */
     @Input()
     public vertical = false;
 
     /**
-     * Propagate dialog properties.
+     *An @Input property that renders today button with custom name.
+     *```html
+     *<igx-datePicker cancelButtonLabel="cancel" todayButtonLabel="Tomorrow"></igx-datePicker>
+     *```
      */
     @Input() public todayButtonLabel: string;
 
+    /**
+     *An @Input property that renders cancel button with custom name.
+     *```html
+     *<igx-datePicker cancelButtonLabel="Close" todayButtonLabel="Today"></igx-datePicker>
+     *```
+     */
     @Input() public cancelButtonLabel: string;
 
+    /**
+     *An @Output property that triggers a function when the `IgxDatePickerComponent` is opened.
```

Comment:
@alexalex972 Please be consistent between the controls. This is event so the description should be: "An event that is emitted when the `IgxDatePickerComponent` is opened."